ledger/light/handle/set_request_test.go

```diff
@@ -0,0 +1,299 @@
+	require.NoError(t, err)
+
+	msg := payload.Meta{
+		Payload: requestBuf,
+	}
+
+	handler := NewSetRequest(emptyDeps(), msg, false)
+
+	err = handler.Present(ctx, f)
+	assert.EqualError(t, err, "object is nil")
+}
+
+func TestSetRequest_FlowWithPassedFlag(t *testing.T) {
+	t.Parallel()
+	ctx := flow.TestContextWithPulse(
+		inslogger.TestContext(t),
+		insolar.GenesisPulse.PulseNumber+10,
+	)
+
+	msg := correctMetaMsg(t)
+
+	t.Run("checkjet procedure returns unknown err", func(t *testing.T) {
+		t.Parallel()
+		f := mockProcedures(t, nil, errors.New("something strange from checkjet"), nil, nil)
+
+		handler := NewSetRequest(emptyDeps(), msg, false)
+		err := handler.Present(ctx, f)
+		assert.EqualError(t, err, "something strange from checkjet")
+	})
+
+	t.Run("passed flag is false and checkjet returns ErrNotExecutor", func(t *testing.T) {
+		t.Parallel()
+		f := mockProcedures(t, nil, proc.ErrNotExecutor, nil, nil)
+
+		handler := NewSetRequest(emptyDeps(), msg, false)
+		err := handler.Present(ctx, f)
+		require.NoError(t, err)
+	})
+
+	t.Run("passed flag is true and checkjet returns ErrNotExecutor", func(t *testing.T) {
+		t.Parallel()
+		f := mockProcedures(t, nil, proc.ErrNotExecutor, nil, nil)
+
+		handler := NewSetRequest(emptyDeps(), msg, true)
+		err := handler.Present(ctx, f)
+		require.Error(t, err)
+		assert.Equal(t, proc.ErrNotExecutor, err)
+	})
+}
+
+func TestSetRequest_ErrorFromWaitHot(t *testing.T) {
+	t.Parallel()
+	ctx := flow.TestContextWithPulse(
+		inslogger.TestContext(t),
+		insolar.GenesisPulse.PulseNumber+10,
+	)
+
+	msg := correctMetaMsg(t)
+
+	t.Run("waithot procedure returns err", func(t *testing.T) {
+		t.Parallel()
+		f := mockProcedures(t, nil, nil, errors.New("error from waithot"), nil)
+
+		handler := NewSetRequest(emptyDeps(), msg, false)
+		err := handler.Present(ctx, f)
+		assert.EqualError(t, err, "error from waithot")
+	})
+
+	t.Run("waithot procedure returns nil err", func(t *testing.T) {
+		t.Parallel()
+		f := mockProcedures(t, nil, nil, nil, nil)
+
+		handler := NewSetRequest(emptyDeps(), msg, false)
+		err := handler.Present(ctx, f)
+		require.NoError(t, err)
+	})
+}
+
+func TestSetRequest_ErrorFromSetRequest(t *testing.T) {
+	t.Parallel()
+	ctx := flow.TestContextWithPulse(
+		inslogger.TestContext(t),
+		insolar.GenesisPulse.PulseNumber+10,
+	)
+
+	msg := correctMetaMsg(t)
+
+	t.Run("setrequest procedure returns err", func(t *testing.T) {
+		t.Parallel()
+		f := mockProcedures(t, nil, nil, nil, errors.New("error from setrequest"))
+
+		handler := NewSetRequest(emptyDeps(), msg, false)
+		err := handler.Present(ctx, f)
+		assert.EqualError(t, err, "error from setrequest")
+	})
+
+	t.Run("setrequest procedure returns nil err", func(t *testing.T) {
+		t.Parallel()
+		f := mockProcedures(t, nil, nil, nil, nil)
+
+		handler := NewSetRequest(emptyDeps(), msg, false)
+		err := handler.Present(ctx, f)
+		require.NoError(t, err)
+	})
+}
+
+func correctMetaMsg(t *testing.T) payload.Meta {
```

Comment:
This fucntion has a generic name, but only creates Request record. It will be a problem when more tests are written.

Reply:
Done. `metaRequestMsg()`